src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/Utils/XBindExpressionParser.cs

```diff
@@ -238,6 +240,48 @@ public override void VisitLiteralExpression(LiteralExpressionSyntax node)
 				_builder.Append(node.ToString());
 			}
 
+			public override void VisitElementAccessExpression(ElementAccessExpressionSyntax node)
+			{
+				if (Failed)
+				{
+					return;
+				}
+
+				if (node.ArgumentList.Arguments.Count != 1 ||
+					node.ArgumentList.Arguments[0].Expression is not LiteralExpressionSyntax index)
+				{
+					throw new Exception("x:Bind indexing expects a single literal argument");
+				}
+
+				Visit(node.Expression);
+
+				var lastType = _lastAccessed.Symbol switch
+				{
+					IPropertySymbol property => property.Type,
+					IFieldSymbol field => field.Type,
+					_ => throw new Exception("Only properties and fields can be indexed.")
+				};
+
+				var expectedIndexerParameterType = index is LiteralExpressionSyntax indexLiteral && indexLiteral.IsKind(SyntaxKind.StringLiteralExpression)
+					? SpecialType.System_String
+					: SpecialType.System_Int32;
+				var indexer = lastType.GetMemberInlcudingBaseTypes(expectedIndexerParameterType, (m, arg) => m is IPropertySymbol { IsIndexer: true } p && p.Parameters[0].Type.SpecialType == arg);
```

Comment:
Note: We could write it as `lastType.GetMemberIncludingBaseTypes(m => /* use expectedIndexerParameterType directly instead of arg*/)` and simplify the signature and implementation of `GetMemberIncludingBaseTypes`. But I wanted to avoid the allocations of display classes (those that the compiler synthesizes when a lambda captures a local)

The allocations should be minor in this case, but I found it easy enough to avoid them, so I went that way.